Initial device registration policy - Preview docs (#12257)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/deviceregistrationpolicy-update.md

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|Policy.ReadWrite.DeviceConfiguration|
|Delegated (personal Microsoft account)|Not supported|
|Application|Not supported|

>**Note:**: For delegated permissions to allow apps to update deviceRegistrationPolicy on behalf of the signed-in user, the tenant administrator must have assigned the user one of the following roles:
* Global administrator
* Cloud device administrator

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
PUT /policies/deviceRegistrationPolicy
```

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body
In the request body, supply a JSON representation of the [deviceRegistrationPolicy](../resources/deviceregistrationpolicy.md) object.

The following table shows the properties that are required when you update the [deviceRegistrationPolicy](../resources/deviceregistrationpolicy.md).

|Property|Type|Description|
|:---|:---|:---|

> |id|String|Read-only.|

Reviewer: Although it may seem descriptive just from the property name, we need to add a description sentence for the properties. Such as, "The identifier of the deviceRegistrationPolicy object."

|displayName|String|Read-only.|
|description|String|Read-only|
|userDeviceQuota|Int32|Specifies the maximum number of devices that a user can have within your organization before blocking new device registrations. Required.|

> |multiFactorAuthConfiguration|multiFactorAuthConfiguration|Specifies the authentication policy for a user to complete registration using Azure AD Join or Azure AD registered within your organization. Possible values are: `notRequired`, `required`, `unknownFutureValue`. Required.|

Reviewer 2: I dont think we specify unknownFutureValue in the docs

> |azureADRegistration|[azureADRegistrationPolicy](../resources/azureadregistrationpolicy.md)|Specifies the authorization policy for controlling registration of new devices using Azure AD registered within your organization. Required. [Learn more](https://docs.microsoft.com/azure/active-directory/devices/overview) about device identities in Azure AD.|

Reviewer 2: This cant be modified by admins if Intune is enabled. Can we document that

> |azureADRegistration|[azureADRegistrationPolicy](../resources/azureadregistrationpolicy.md)|Specifies the authorization policy for controlling registration of new devices using Azure AD registered within your organization. Required. [Learn more](https://docs.microsoft.com/azure/active-directory/devices/overview) about device identities in Azure AD.|

Reviewer: ```suggestion
|azureADRegistration|[azureADRegistrationPolicy](../resources/azureadregistrationpolicy.md)|Specifies the authorization policy for controlling registration of new devices using Azure AD registered within your organization. Required. For more information, see [What is a device identity?](/azure/active-directory/devices/overview).|
```

> |azureADJoin|[azureAdJoinPolicy](../resources/azureadjoinpolicy.md)|Specifies the authorization policy for controlling registration of new devices using Azure AD Join within your organization. Required. [Learn more](https://docs.microsoft.com/azure/active-directory/devices/overview) about device identities in Azure AD.|

Reviewer: ```suggestion
|azureADJoin|[azureAdJoinPolicy](../resources/azureadjoinpolicy.md)|Specifies the authorization policy for controlling registration of new devices using Azure AD Join within your organization. Required. For more information, see [What is a device identity?](/azure/active-directory/devices/overview).|
```

* Global administrator
* Cloud device administrator

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
PUT /policies/deviceRegistrationPolicy
```

## Request headers

|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body

In the request body, supply a JSON representation of the [deviceRegistrationPolicy](../resources/deviceregistrationpolicy.md) object.

The following table shows the properties that are required when you update the [deviceRegistrationPolicy](../resources/deviceregistrationpolicy.md).

|Property|Type|Description|
|:---|:---|:---|
|id|String| This string is read-only and is always set to `deviceRegistrationPolicy`. |
|displayName|String| This string is read-only and is always set to `Device Registration Policy`. |
|description|String| This string is read-only and is always set to `Tenant-wide policy that manages intial provisioning controls using quota restrictions, additional authentication and authorization checks`. |

> |userDeviceQuota|Int32|Specifies the maximum number of devices that a user can have within your organization before blocking new device registrations. Required.|
> |multiFactorAuthConfiguration|multiFactorAuthConfiguration|Specifies the authentication policy for a user to complete registration using Azure AD Join or Azure AD registered within your organization. Possible values are: `notRequired` or `required`. Required.|

Reviewer 3: Kindly confirm. Neither property seemed required when I tested.
```suggestion
|userDeviceQuota|Int32|Specifies the maximum number of devices that a user can have within your organization before blocking new device registrations. |
|multiFactorAuthConfiguration|multiFactorAuthConfiguration|Specifies the authentication policy for a user to complete registration using Azure AD Join or Azure AD registered within your organization. Possible values are: `notRequired` or `required`. |
```

## Response

If successful, this method returns a `200 OK` response code and an updated [deviceRegistrationPolicy](../resources/deviceregistrationpolicy.md) object in the response body.

## Examples

### Request

<!-- {
  "blockType": "request",
  "name": "update_deviceregistrationpolicy"
}
-->
``` http
PUT https://graph.microsoft.com/beta/deviceRegistrationPolicy
Content-Type: application/json

{
    "id": "deviceRegistrationPolicy",
    "displayName": "Device Registration Policy",
    "description": "Tenant-wide policy that manages intial provisioning controls using quota restrictions, additional authentication and authorization checks",
    "userDeviceQuota": 50,
    "multiFactorAuthConfiguration": "0",
    "azureADRegistration": {
        "appliesTo": "1",
        "isAdminConfigurable": false,
        "allowedUsers": [],
        "allowedGroups": []
    },
    "azureADJoin": {
        "appliesTo": "1",
        "isAdminConfigurable": true,
        "allowedUsers": [],
        "allowedGroups": []
    }
}
```

### Response

> **Note:** The response object shown here might be shortened for readability.

Reviewer 3: ```suggestion
>**Note:** The response object shown here might be shortened for readability.
```
